### PR TITLE
AWS for Fluent Bit 2.32.3 with Input Grace period fix

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*

Diff: https://github.com/PettitWesley/fluent-bit/pull/33/commits

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 
+### 2.32.3
+This release includes:
+* Fluent Bit [1.9.10](https://github.com/fluent/fluent-bit/tree/v1.9.10)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.4
+* Amazon Kinesis Streams for Fluent Bit 1.10.2
+* Amazon Kinesis Firehose for Fluent Bit 1.7.2
+
+Compared to `2.31.12` this release adds:
+* Fix - During shutdown, the `Grace` setting controls how long Fluent Bit waits after receiving a SIGTERM before exiting. Previously, input ingestion was paused one second after shutdown; with this change Fluent Bit stops acceptings new logs when one half of the `Grace` period time has elapsed.
```
Comment (PettitWesley) [inline CHANGELOG.md:11]: also set default grace to 30

and check buffers on shutdown

**Review by ShelbyZ — APPROVED** [review]
LGTM

**Comment by everhardt** [comment]
@sparrc Why did you close this PR? The feature it brings (specifically `grace_input`) looks very useful for an issue we have that is similar to #856: a SIGSEGV on the main container causes it to exit, but the last logging lines are never pushed to our loki destination.

**Comment by swapneils** [comment]
@everhardt We've made some changes to our deployment process that invalidate the specific code in this change, which is why we closed the PR.

Separately, this change requires more internal testing to confirm it solves the issue in all cases; there are some anomalies in that testing which we are currently looking into. We don't intend to publish the change until we've confirmed that it works properly.

**Pull request closed without merging**